### ISSUE TITLE
Ruby on Rails: Replace Flickr API Project

### ISSUE THREAD

**Issue by AhmedTheGreatest** [TheOdinProject/curriculum#28285]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Problem
Many users have reported encountering issues with the Flickr API project within TheOdinProject curriculum. Specifically, problems include difficulties with signing in, logging in, and frequent "Account creation failed" errors.

### Issues with Flickr
- **Account Creation and Login Problems:** Users often face errors when attempting to create accounts or log in to Flickr.
- **Outdated Platform:** Flickr is considered somewhat outdated, which may affect the learning experience and the relevance of skills acquired.

### Solution
I suggest that we can use some other API, for example Imgur's API. Its modern, user-friendly, and more widely used.

### Path

Ruby / Rails

### Lesson Url

https://www.theodinproject.com/lessons/ruby-on-rails-flickr-api

### (Optional) Discord Name

tornado_77

### (Optional) Additional Comments

I will see if I can help depending on where this issue goes.

**Comment by CouchofTomato** [comment]
I'm not against this but it's a meaty bit of work.

**Comment by AhmedTheGreatest** [comment]
Yeah I agree, it is alot of work. But I think it will be worth it. I myself am stuck on that project.

**Comment by AhmedTheGreatest** [comment]
I will try to do some research on how we could change the lesson (or if we even should) to Imgur API. I am also stuck on this project right now.

**Comment by weeezik** [comment]
@AhmedTheGreatest Find anything? Still want help?

**Comment by AhmedTheGreatest** [comment]
I didn't really look into it much. But I have seen more and more people having issues with Flickr on the discord server.

**Comment by weeezik** [comment]
I'm not on the project yet - if the inspiration is still there we can circle back to it. Might take me a bit to get there. Let me know if you want help on it at any point, I can help. I just don't have the capacity to the task on fully at the moment.

**Comment by AhmedTheGreatest** [comment]
@weeezik Sure. I am currently not working on this at all. But yeah the problem still persists. So any help on this would be great.

**Comment by JoshDevHub** [comment]
New issue that bumps this in priority a lot: Flickr now requires users to be subscribed to a Flickr PRO premium account in order to generate API keys. As opposed to the mysterious account creation issue that only effected some users, this will affect *all* users.

Link to discord discussion: https://discordapp.com/channels/505093832157691914/514204607770001411/1351113608364560404

![Image](https://github.com/user-attachments/assets/2107e009-1459-4214-827b-90fdbb8513eb)

**Comment by AhmedTheGreatest** [comment]
Hmm yeah, that does make this issue much more prevalent then. It basically means that one of the projects in the curriculum is now locked behind a user being on a paid subscription to a third-party application.

**Comment by NikEmman** [comment]
Hello there. I have done some research for a replacement. The best I came up with is [Pexels](https://www.pexels.com/).

Features free account creation and free API key generation, with a limit of 20.000 requests per month, plenty for the curriculum's needs. Should I go ahead and write some instructions for setting up an account, API key acquisition etc?, in the style of the current lesson instructions for Flickr?

Here are the Pexel API [docs](https://www.pexels.com/api/documentation/#statistics).

There is also a gem that simplifies the API calls, it's the [pexels-ruby](https://github.com/pexels/pexels-ruby) gem.

**Comment by CouchofTomato** [comment]
Seems our best bet @NikEmman

**Comment by NikEmman** [comment]
I have re-written the lesson with instructions to use Pexel instead of Flickr. On my  curiculum fork I deleted the old `project_flickr_api.md` and added a new file, `project_pexel_api.md`.

However now I am unsure on how to proceed. Should I just go ahead and do a PR? I couldn't figure out how the md files are pulled by the main TheOdinProject app to be rendered as content. Does it expect the old file name or will it render the apis folder contents?

@CouchofTomato

**Comment by JoshDevHub** [comment]
@NikEmman Thank you for your interest in contributing on this.

A couple things:
1. Make sure you ask to be assigned an issue before beginning work on it. This just helps eliminate some situation where 2 or more people work on the same thing, which inevitably leads to someone wasting their time. Given your interest in working on this, I'll assign you now. Definitely let me know if you have any other questions about the content or process.
2. Don't delete the old file. Instead, move it to `archive/`. To actually get the new project on the site, you'll need a PR to the [main site repo](https://github.com/TheOdinProject/theodinproject) that removes the old lesson from the Rails course lesson fixtures and adds the new one. But you don't have to worry about that just yet. Assuming you have some content that's ready to go, you can go ahead and PR that. Once it's reviewed and approved, then we can talk more about the process for getting it to actually be on the site.

**Comment by NikEmman** [comment]
Thanks @JoshDevHub .  I picked the issue because it was flagged as stale, thus assumed no-one was on it. You are absolutely right, should have waited for the assignment.  Hopefully I didn't waste anyone's time.

**Comment by JoshDevHub** [comment]
Oh you didn't waste anyone's time. I edited my post after noticing the above conversation with timato. You did everything fine 😃

**Comment by Claire135** [comment]
The API key is now only available for pro users - you have to pay to complete this element of the course.
